engine/src/main/java/com/datatorrent/stram/stream/InlineStream.java

```diff
@@ -80,14 +88,29 @@ public void teardown()
   public void put(Object tuple)
   {
     try {
-      super.put(tuple);
+      reservoir.put(tuple);
+      if (!(tuple instanceof Tuple)) {
```

Comment:
I don't see any difference in InlineStream performance before and after the change.